modules/core/src/test/scala/org/tessellation/infrastructure/trust/TrustStorageUpdaterSuite.scala

```diff
@@ -0,0 +1,163 @@
+import org.tessellation.schema.trust.SnapshotOrdinalPublicTrust
+import org.tessellation.sdk.domain.gossip.Gossip
+import org.tessellation.sdk.domain.trust.storage.OrdinalTrustMap.{ordinalTrustUpdateDelay, ordinalTrustUpdateInterval}
+import org.tessellation.sdk.domain.trust.storage.{OrdinalTrustMap, TrustMap}
+
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.NonNegative
+import io.circe.Encoder
+import org.scalacheck.Gen
+import weaver.SimpleIOSuite
+import weaver.scalacheck.Checkers
+
+object TrustStorageUpdaterSuite extends SimpleIOSuite with Checkers {
+  def mkMockGossip[B](spreadRef: Ref[IO, List[B]]): Gossip[IO] = new Gossip[IO] {
+    override def spread[A: TypeTag: Encoder](rumorContent: A): IO[Unit] =
+      spreadRef.update(rumorContent.asInstanceOf[B] :: _)
+
+    override def spreadCommon[A: TypeTag: Encoder](rumorContent: A): IO[Unit] =
+      IO.raiseError(new Exception("spreadCommon: Unexpected call"))
+  }
+
+  test("trust storages are updated and gossiped") {
+    val minOrdinalValue = (2L * ordinalTrustUpdateInterval) + ordinalTrustUpdateDelay + 1L
+    val maxOrdinalValue = (3L * ordinalTrustUpdateInterval) - 1L
+
+    val gen = for {
+      trust <- Gen.mapOfN(4, genPeerTrustInfo)
+      delta <- Gen.chooseNum(minOrdinalValue, maxOrdinalValue)
+      ordinal = Refined.unsafeApply[Long, NonNegative](delta)
+    } yield (TrustMap.empty.copy(trust = trust), SnapshotOrdinal(ordinal))
+
+    forall(gen) {
+      case (trust, ordinal) =>
+        for {
+          store <- TrustStorage.make(trust)
+          gossipped <- Ref.of(List.empty[SnapshotOrdinalPublicTrust])
+          mockGossip = mkMockGossip(gossipped)
+          updater = TrustStorageUpdater.make(ordinal.some.pure, mockGossip, store)
+
+          firstCurrent <- store.getCurrentOrdinalTrust
+          firstNext <- store.getNextOrdinalTrust
+          expectedFirstCurrent = OrdinalTrustMap.empty
+          expectedFirstNext = OrdinalTrustMap.initial
+
+          _ <- store.updateNext(SnapshotOrdinal(ordinalTrustUpdateInterval))
+          _ <- updater.update
+
+          secondCurrent <- store.getCurrentOrdinalTrust
+          secondNext <- store.getNextOrdinalTrust
+          expectedSecondCurrent = OrdinalTrustMap.initial.copy(trust = trust)
+          expectedSecondNext = OrdinalTrustMap.advanceInterval(OrdinalTrustMap.initial).copy(trust = trust)
+          wasGossipped <- gossipped.get
+          expectedGossip = SnapshotOrdinalPublicTrust(
+            SnapshotOrdinal(2000L),
+            trust.toPublicTrust
+          )
+        } yield
+          expect.all(
+            expectedFirstCurrent === firstCurrent,
+            expectedFirstNext === firstNext,
+            expectedSecondCurrent === secondCurrent,
+            expectedSecondNext === secondNext,
+            wasGossipped === List(expectedGossip)
+          )
+    }
+  }
+
+  test("trust storages are not updated and gossiped") {
```

Comment:
what is the difference between this and the previous test? it would be good to add `when` to the test. like:
```
trust storages are updated and gossiped after next trust has been updated
```
or something like that.

also, I'd advice to split these tests into two. one checking for updates and second one checking for gossiping. and maybe write some function for initialization or extracting store values, as it's already quite big

Reply:
The previous tests shows that if the ordinal trust storages are updated, then they are gossiped. This test shows the opposite, if the ordinal trust storages are not updated, nothing is gossiped.

Reply:
ok, so it should be enough to check this:
```scala
test("trust storages are updated") {
    val minOrdinalValue = (2L * 1000L) + 500L + 1L
    val maxOrdinalValue = (3L * 1000L) - 1L

    val gen = for {
      trust <- Gen.mapOfN(4, genPeerTrustInfo)
      delta <- Gen.chooseNum(minOrdinalValue, maxOrdinalValue)
      ordinal = Refined.unsafeApply[Long, NonNegative](delta)
    } yield (TrustMap.empty.copy(trust = trust), SnapshotOrdinal(ordinal))

    forall(gen) {
      case (trust, ordinal) =>
        for {
          store <- mkTrustStorage(trust)
          gossipped <- Ref.of(List.empty[SnapshotOrdinalPublicTrust])
          mockGossip = mkMockGossip(gossipped)
          updater = TrustStorageUpdater.make(ordinal.some.pure, mockGossip, store)

          _ <- store.updateNext(SnapshotOrdinal(1000L))
          _ <- updater.update

          current <- store.getCurrentOrdinalTrust
          next <- store.getNextOrdinalTrust
          expectedCurrent = OrdinalTrustMap.empty.copy(ordinal = SnapshotOrdinal(1000L), trust = trust)
          expectedNext = OrdinalTrustMap.empty.copy(ordinal = SnapshotOrdinal(2000L), trust = trust)
         
        } yield expect.eql((expectedCurrent, expectedNext), (current, next))
    }
  }

test("snapshot ordinal public trust is gossiped after update") {
    val minOrdinalValue = (2L * 1000L) + 500L + 1L
    val maxOrdinalValue = (3L * 1000L) - 1L

    val gen = for {
      trust <- Gen.mapOfN(4, genPeerTrustInfo)
      delta <- Gen.chooseNum(minOrdinalValue, maxOrdinalValue)
      ordinal = Refined.unsafeApply[Long, NonNegative](delta)
    } yield (TrustMap.empty.copy(trust = trust), SnapshotOrdinal(ordinal))

    forall(gen) {
      case (trust, ordinal) =>
        for {
          store <- mkTrustStorage(trust)
          gossipped <- Ref.of(List.empty[SnapshotOrdinalPublicTrust])
          mockGossip = mkMockGossip(gossipped)
          updater = TrustStorageUpdater.make(ordinal.some.pure, mockGossip, store)

          _ <- store.updateNext(SnapshotOrdinal(1000L))
          _ <- updater.update

          wasGossipped <- gossipped.get
          expectedGossip = SnapshotOrdinalPublicTrust(
            SnapshotOrdinal(2000L),
            trust.toPublicTrust
          )
        } yield expect.eql(List(expectedGossip), wasGossiped)
    }
  }
```

and the same for the other case

Reply:
also, you can export this:
```scala
    val gen = for {
      trust <- Gen.mapOfN(4, genPeerTrustInfo)
      delta <- Gen.chooseNum(minOrdinalValue, maxOrdinalValue)
      ordinal = Refined.unsafeApply[Long, NonNegative](delta)
    } yield (TrustMap.empty.copy(trust = trust), SnapshotOrdinal(ordinal))
```
into function that receives `minOrdinalValue` and `maxOrdinalValue`, to remove duplicated code

Reply:
Not sure if this is just a stylistic request for change or impacts the test but two test cases here seem fine to me. The generator though can be deduplicated.

Reply:
De-duplicated the generator and replace my tests with yours, @marcinwadon .